sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ShowTablesExec.scala

```diff
@@ -33,14 +44,37 @@ case class ShowTablesExec(
     output: Seq[Attribute],
     catalog: TableCatalog,
     namespace: Seq[String],
-    pattern: Option[String]) extends V2CommandExec with LeafExecNode {
+    pattern: Option[String],
+    isExtended: Boolean = false,
+    partitionSpec: Option[TablePartitionSpec] = None) extends V2CommandExec with LeafExecNode {
```

Comment:
`TablePartitionSpec` is legacy one, can't you use `ResolvedPartitionSpec`? For example, see https://github.com/apache/spark/blob/0494dc90af48ce7da0625485a4dc6917a244d580/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ShowPartitionsExec.scala#L36

Reply:
In fact, `ResolvedPartitionSpec` cannot be used here, because the table(`Pattern` match may have multiple tables) has not been determined.
The scene is different from `ShowPartitionExec`.

Reply:
two chooses:
1.AstBuilder(visitShowTableExtended) don't use UnresolvedPartitionSpec, instead, use Map<String, String>. Because of the lack of `decisive table`, it is impossible to resolve to `ResolvedPartitionSpec`.
2.AstBuilder(visitShowTableExtended) use UnresolvedPartitionSpec, have to add a subclass(eg: ExtractPartitionSpec, it's `resolved` is true)
The root cause is that the table is `non-deterministic` (Pattern match may have multiple tables) in the ShowTableExtended.

Reply:
Done